spring-boot-1

ShemsheyMaxim opened the pull request:
added classes Role and User,
added package repository,
added interface RoleRepository and UserRepository,
added package service,
added interface RoleService and class RoleServiceImpl,
added interface UserService and class UserServiceImpl,
added package exception and class DataProcessingException,
added package controller and class InjectData

ShemsheyMaxim committed:
added package model,
added classes Role and User,
added package repository,
added interface RoleRepository and UserRepository,
added package service,
added interface RoleService and class RoleServiceImpl,
added interface UserService and class UserServiceImpl,
added package exception and class DataProcessingException,
added package controller and class InjectData

Sashk52 commented on src/main/java/com/springboot/onlinebanking/repository/RoleRepository.java:
```diff
@@ -0,0 +1,11 @@
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    @Query("SELECT r FROM Role r WHERE r.roleName = :roleName")
+    Optional<Role> getRoleByRoleName(String roleName);
```
....getRoleByRoleName( @Param("roleName")String roleName));

I think you have add @Param here.

boroda4436 replied:
@Sashk52 `@Param` is not required here

boroda4436 commented on src/main/java/com/springboot/onlinebanking/repository/RoleRepository.java:
```diff
@@ -0,0 +1,11 @@
+import org.springframework.data.jpa.repository.Query;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    @Query("SELECT r FROM Role r WHERE r.roleName = :roleName")
```
@ShemsheyMaxim Can we avoid using `@Query` here?

ShemsheyMaxim replied:
done

boroda4436 commented on src/main/java/com/springboot/onlinebanking/repository/UserRepository.java:
```diff
@@ -0,0 +1,7 @@
+import com.springboot.onlinebanking.model.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface UserRepository extends JpaRepository<User, Long> {
```
@ShemsheyMaxim Do we need to add `findByNumber`?

ShemsheyMaxim replied:
done

ShemsheyMaxim committed:
changed method getRoleByRoleName() in RoleRepository class,
added method findUserByPhoneNumber() to UserRepository interface,
added method findUserByUserNumber() to UserService interface and UserServiceImpl class
